pkg/client/unversioned/helper_test.go

```diff
@@ -191,3 +192,81 @@ func TestHelperGetServerAPIVersions(t *testing.T) {
 		t.Errorf("expected %v, got %v", e, a)
 	}
 }
+
+func TestSetsCodec(t *testing.T) {
+	testCases := map[string]struct {
+		Err    bool
+		Prefix string
+		Codec  runtime.Codec
+	}{
+		testapi.Default.Version(): {false, "/api/" + testapi.Default.Version() + "/", testapi.Default.Codec()},
+		"invalidVersion":          {true, "", nil},
+	}
+	for version, expected := range testCases {
+		client, err := New(&Config{Host: "127.0.0.1", GroupVersion: &unversioned.GroupVersion{Version: version}})
+		switch {
+		case err == nil && expected.Err:
+			t.Errorf("expected error but was nil")
+			continue
+		case err != nil && !expected.Err:
+			t.Errorf("unexpected error %v", err)
+			continue
+		case err != nil:
+			continue
+		}
+		if e, a := expected.Prefix, client.RESTClient.baseURL.Path; e != a {
+			t.Errorf("expected %#v, got %#v", e, a)
+		}
+		if e, a := expected.Codec, client.RESTClient.Codec; e != a {
+			t.Errorf("expected %#v, got %#v", e, a)
+		}
+	}
+}
+
+func TestRESTClientRequires(t *testing.T) {
+	if _, err := RESTClientFor(&Config{Host: "127.0.0.1", Codec: testapi.Default.Codec()}); err == nil {
+		t.Errorf("unexpected non-error")
+	}
+	if _, err := RESTClientFor(&Config{Host: "127.0.0.1", GroupVersion: testapi.Default.GroupVersion()}); err == nil {
+		t.Errorf("unexpected non-error")
+	}
+	if _, err := RESTClientFor(&Config{Host: "127.0.0.1", GroupVersion: testapi.Default.GroupVersion(), Codec: testapi.Default.Codec()}); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+}
+
+func TestValidatesHostParameter(t *testing.T) {
+	testCases := []struct {
+		Host   string
+		Prefix string
+
+		URL string
+		Err bool
+	}{
+		{"127.0.0.1", "", "http://127.0.0.1/" + testapi.Default.Version(), false},
+		{"127.0.0.1:8080", "", "http://127.0.0.1:8080/" + testapi.Default.Version(), false},
+		{"foo.bar.com", "", "http://foo.bar.com/" + testapi.Default.Version(), false},
+		{"http://host/prefix", "", "http://host/prefix/" + testapi.Default.Version(), false},
+		{"http://host", "", "http://host/" + testapi.Default.Version(), false},
+		{"http://host", "/", "http://host/" + testapi.Default.Version(), false},
+		{"http://host", "/other", "http://host/other/" + testapi.Default.Version(), false},
```

Comment:
Trailing slashes were removed from the URL because it was being added in the RESTClient constructor. Has nothing to really do with the main verification logic.